arroyo-rpc/src/types.rs

```diff
@@ -296,3 +299,284 @@ pub struct Connector {
     pub hidden: bool,
     pub connection_config: Option<String>,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct ConnectionProfile {
+    pub id: String,
+    pub name: String,
+    pub connector: String,
+    pub config: String,
```

Comment:
Good idea. Are you suggesting that we get rid of the `Connector` trait's associated types `ConfigT` and `TableT`?

Reply:
I think that's still useful because it turns them into concrete types that the connectors can work with. But instead of using strings as the inputs to the ErasedConnector functions, we can use Values (and throughout, avoiding a bunch of serialization/deserialization).

Reply:
Although we can rename ConfigT to ProfileT or something

Reply:
Cool, done. One thing I wasn't sure about was `EmptyConfig`. Since `serde_json::Value::Null` doesn't parse into that, I had to change `EmptyConfig` to be a type alias for `serde_json::Value`, which works, but is a little unclear because it should always be the `Null` variant. Do you have any better ideas?

Reply:
`EmptyConfig` is actually `{}`, not null so I think it should work correctly as is

Reply:
Right, but an empty json is deserialized as `serde_json::Value::Null`, so this causes a panic:

```let _: EmptyConfig = serde_json::from_value(serde_json::Value::Null).unwrap();```

Reply:
``` Invalid connector config for 110: Error("invalid type: null, expected struct EmptyConfig", line: 0, column: 0)```

Reply:
Never mind, you were right, I've put it back how it was.